src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/AutoFlushingNerdbankStream.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.VisualStudio.LanguageServerClient.Razor
+{
+    // Intended for use with a party from a Nerdbank FullDuplexStream:
+    // https://github.com/AArnott/Nerdbank.Streams/blob/master/doc/FullDuplexStream.md
+    internal class AutoFlushingNerdbankStream : Stream
+    {
+        private readonly Stream _inner;
+
+        public AutoFlushingNerdbankStream(Stream inner)
+        {
+            _inner = inner;
+        }
+
+        public override bool CanRead => _inner.CanRead;
+
+        public override bool CanSeek => _inner.CanSeek;
+
+        public override bool CanWrite => _inner.CanWrite;
+
+        public override long Length => _inner.Length;
+
+        public override long Position { get => _inner.Position; set => _inner.Position = value; }
+
+        public override long Seek(long offset, SeekOrigin origin) => _inner.Seek(offset, origin);
+
+        public override void SetLength(long value) => _inner.SetLength(value);
+
+        // We ensure the Read/Write/Flush calls happen in a synchronous manner to avoid
+        // concurrent read/write exceptions resulting in crashes of the language server.
+        public override void Flush() => FlushAsync(CancellationToken.None).GetAwaiter().GetResult();
```

Comment:
The comment makes sense but I still don't understand how this is different from calling `_inner.Flush()`. Maybe I'm missing something basic here.

Reply:
Does https://github.com/dotnet/aspnetcore-tooling/pull/1861#discussion_r419741825 answer your question? If not, I'd be happy to expand / clarify 😄

Reply:
It does. I missed it originally. Thanks